plugins/builds/update.js

```diff
@@ -7,6 +7,30 @@ const schema = require('screwdriver-data-schema');
 const { EXTERNAL_TRIGGER } = schema.config.regex;
 const idSchema = joi.reach(schema.models.job.base, 'id');
 
+/**
+ * Determine if this build is FIXED build or not.
+ * @method isFixedBuild
+ * @param  build
+ * @param  jobFactory
+ */
+async function isFixedBuild(build, jobFactory) {
```

Comment:
Is it guaranteed that when we get to this function, build status is `SUCCESS`?

If not, I think it might be good to add a check if build status is `SUCCESS` before fetching failureBuild/successBuild since it seems like a lot of additional calls to the DB
should we check if new build status is SUCCESS first

Reply:
I fixed it.